projects/button_challenge/src/main.c

```diff
@@ -0,0 +1,64 @@
+#include <stdint.h>
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "wait.h"
+
+// Callback function
+static void prv_adc_callback(const GpioAddress *address, void *context) {
+  GpioAddress *adc_address = context;
+  uint16_t adc_data = 0;
+
+  adc_read_raw_pin(*adc_address, &adc_data);
+  LOG_DEBUG("ADC data: %d\n", adc_data);
+}
+
+// Main
+int main() {
+  interrupt_init();
+  gpio_init();
+  gpio_it_init();
+
+  // Addresses
+  const GpioAddress button_address = { .port = GPIO_PORT_B, .pin = 2 };
```

Comment:
For formatting, we normally put a comma at the end
(eg .pin = 2,) instead of just (.pin = 2)
Please fix this for all the structs